### PR TITLE
[chore] use ReportComponentStatus

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Lint failures

**Comment by bogdandrutu** [comment]
Failing tests in the extensions UT, probably related https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7242682831/job/19728444138?pr=29907

File: extension/observer/ecsobserver/extension.go [hunk 1]
```diff
@@ -15,15 +15,16 @@ var _ extension.Extension = (*ecsObserver)(nil)
 
 // ecsObserver implements component.ServiceExtension interface.
 type ecsObserver struct {
-	logger *zap.Logger
-	sd     *serviceDiscovery
+	logger            *zap.Logger
```
Comment (bogdandrutu) [inline extension/observer/ecsobserver/extension.go:18]: Don't need logger, it is in `telemetrySettings.Logger`.

**Review by bogdandrutu — APPROVED** [review]

File: extension/observer/ecsobserver/extension.go (hunk 1 above)
Comment (atoulme) [reply extension/observer/ecsobserver/extension.go]: Removed, thanks

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**